_posts/2019-06-25-code-reviews-and-your-company-goal.markdown

```diff
@@ -0,0 +1,98 @@
+
+Now we get to the meat of the problem. Are code reviews productive for the company? They take precious engineering time and use it to look at code that is already written. Wouldn't it be more productive to use that time to write more code? The answer to this is no, but it is extremely important that the code reviews are implemented in an efficient manner. When done wrong even though the intentions are good, they can have very negative results.
+
+## A bad example
+
+The CTO of Acme inc is obsessed with code quality and sharing knowledge between the engineering team. So much so that the CTO proposes a process that requires every code change to be reviewed by all developers in the engineering department (more than 10 people!).
+
+What are the impacts of this new process to the company?
+
+- This new process will increase code quality as all these reviewers will find more issues with the code than a single or two reviewers ever could. 👍
+- This new process will also increase knowledge sharing. Every engineer on the team will see every change going out. 👍
+- But this process will make feature velocity grind to a halt. All the engineers will be too busy reviewing all the code! 👎
+
+The company will struggle if releasing new features to the customers will require this many reviews, so even though the intention was a nobel one, it will actually hurt the company's goal.
+
+> The road to hell is paved with good intentions.
+
+I have seen a much milder version of this first hand and it wasn't at all obvious that we were doing something counterproductive. We were just focusing on the wrong things when reviewing code, making code reviews too rigid and making the process take too much engineering time without bringing enough value.
+
+So to properly understand how to structure our code review process, lets look at all the main benefits from the list above and see how much value they bring to the company's goal. This will help us figure out what we need to focus on the most when reviewing code.
+
+## 1. Catch code defects before the code is shipped to QA or prod
+
+During my career I have heard the following phrase multiple times:
+
+> Code reviews aren't there to catch bugs, we have the QA team for that.
+
+But if you are looking at code reviews from the standpoint of the value that they bring to the company, detecting issues early is probably the biggest impact that a code review can make. It's definitely the only thing that the users will notice!
+
+Even if you have an amazing QA team that makes sure that bugs don't get deployed to production, there is still value of finding bugs during the code review process. Fixing the bug found during code review will be much easier and quicker as it will require less people to be involved. You might not even need a Jira ticket for it! This will increase your turnaround time and make sure users get your new feature sooner. Quick turnaround time adds value to your company.
```

Comment:
```suggestion
Even if you have an amazing QA team that makes sure that bugs don't get deployed to production, there is still the value of finding bugs during the code review process. Fixing the bug found during code review will be much easier and quicker as it will require fewer people to be involved. You might not even need a Jira ticket for it! This will increase your turnaround time and make sure users get your new feature sooner. Quick turnaround time adds value to your company.
```